src/bitops.c

```diff
@@ -30,7 +30,15 @@
 
 #include "server.h"
 #ifdef HAVE_AVX2
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunknown-pragmas"
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+
 #include <immintrin.h>
+
+#pragma clang diagnostic pop
+#pragma GCC diagnostic pop
```

Comment:
```suggestion
/* Define __MM_MALLOC_H to prevent importing the memory aligned
 * allocation functions, which we don't use. */
#define __MM_MALLOC_H
#include <immintrin.h>
```
I don't know if this is a better approach, but this is what I put in hyperloglog.c. My thinking was I wanted to make sure we aren't calling the malloc from the intrinsics, and this will prevent them from being used. The downside is that it felt hacky to me.

Reply:
Your approach makes sense to me. That way is guaranteed that malloc isn't used..